src/ApiService/ApiService/Functions/QueueFileChanges.cs

```diff
@@ -47,7 +53,39 @@ public class QueueFileChanges {
         await FileAdded(_log, fileChangeEvent, lastTry);
     }
 
-    private async Async.Task FileAdded(ILogTracer log, JsonDocument fileChangeEvent, bool failTaskOnTransientError) {
+    [Function("QueueFileChangesPoison")]
+    public async Async.Task Poison(
+        [QueueTrigger(QueueFileChangesPoisonQueueName, Connection = "AzureWebJobsStorage")] string msg) {
+        var fileChangeEvent = JsonSerializer.Deserialize<JsonDocument>(msg, EntityConverter.GetJsonSerializerOptions());
+
+        _ = fileChangeEvent ?? throw new ArgumentException("Unable to parse queue trigger as JSON");
+
+        // check type first before calling Azure APIs
+        const string eventType = "eventType";
+        if (!fileChangeEvent.RootElement.TryGetProperty(eventType, out var eventTypeElement)
+            || eventTypeElement.GetString() != "Microsoft.Storage.BlobCreated") {
+            return;
+        }
+
+        const string topic = "topic";
+        if (!fileChangeEvent.RootElement.TryGetProperty(topic, out var topicElement)
+            || !_storage.CorpusAccounts().Contains(new ResourceIdentifier(topicElement.GetString()!))) {
+            return;
+        }
+
+        try {
+            // Setting isLastRetryAttempt to false will rethrow any exceptions
+            // With the intention that the azure functions runtime will handle requeing
+            // the message for us. The difference is for the poison queue, we're handling the
+            // requeuing ourselves because azure functions doesn't support retry policies
+            // for queue based functions.
+            await FileAdded(_log, fileChangeEvent, false);
```

Comment:
Stylistic suggestion: I prefer naming bare arguments like this so it is clear what they are doing:
```suggestion
            await FileAdded(_log, fileChangeEvent, isLastRetryAttempt: false);
```